### PR TITLE
feat: add complex/parse-float32

### LINKED ISSUE #1332: [RFC]: add `@stdlib/complex/parse-float32`
### Description

This RFC proposes adding the package `@stdlib/complex/parse-float32`.

The package should be similar in structure to [`@stdlib/complex/reviver-float32`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/complex/reviver-float32).

The goal is to parse a complex number string (e.g., `1 + 8j`) and return a `Complex64` instance.

```javascript
function parseComplex64( str ) { ... }
```

Package: `@stdlib/complex/parse-float32`
Alias: `parseComplex64`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #1332 .

## Description

> What is the purpose of this pull request?

This pull request:
Adds the package @stdlib/complex/parse-float32.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1332 

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Tests:
Screenshot 2024-02-29 230900

Benchmarks
Screenshot 2024-02-29 230841

Examples
Screenshot 2024-02-29 230624

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,188 +0,0 @@
-#/
```
Comment (Pranavchiku) [inline .gitignore]: can you please revert this

Comment (Userf1995) [reply .gitignore]: Hi Pranav I reverted this and also What are the changes I have to do, I changed it now please check

File: lib/node_modules/@stdlib/complex/parse-float32/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+var parseComplex64 = require( '@stdlib/complex/parse-float32' );
+```
+
+#### parseComplex64( str )
+
+Parse a string representation of a 64-bit [complex number][@stdlib/complex/float32].
+
+```javascript
+var real = require( '@stdlib/complex/real' );
+var imag = require( '@stdlib/complex/imag' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/complex/parse-float32/README.md:50]:
```suggestion
var parseComplex64 = require( '@stdlib/complex/parse-float32' );
var real = require( '@stdlib/complex/real' );
var imag = require( '@stdlib/complex/imag' );
```

File: lib/node_modules/@stdlib/complex/parse-float32/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    out: Complex64
+        64-bit complex number.
+
+    Examples
+    --------
+    > var str = '5 + 3i';
+    > var z = {{alias}}( str )
+    <Complex64>
+
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/complex/parse-float32/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/complex/parse-float32/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var Complex64 = require( '@stdlib/complex/float32' );
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/complex/parse-float32/lib/main.js]:
```suggestion
var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
var Complex64 = require( '@stdlib/complex/float32' );
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Looks good to me! I didn't went much into `lib/main.js` overall it is good. Requested few changes, once addressed this will be ready to merge.

File: lib/node_modules/@stdlib/complex/parse-float32/lib/main.js (hunk 4 above)
Comment (Userf1995) [reply lib/node_modules/@stdlib/complex/parse-float32/lib/main.js]: I've made all the changes you mentioned

**Review by Pranavchiku — APPROVED** [review]
Looks good to me. Thanks @Rejoan-Sardar !

**Pull request merged**